lib/services/database/user_repository_service.dart

```diff
@@ -44,6 +44,13 @@ class UserRepositoryService {
         .get();
     return query.docs.isNotEmpty;
   }
+
+  /// This method will add [points] to the user with id [uid]
+  Future<void> addPoints(UserIdFirestore uid, int points) async {
+    final user = await getUser(uid);
+    final updatedData = user.data.addPointstoDbData(points);
+    await _collectionRef.doc(uid.value).update(updatedData);
```

Comment:
Good job in adapting the repository for the addition of centauri points :+1: 

I just have a little suggestion to use a transaction to make the update atomic.
This ensures that if the user document is modified by something else at the same time (which could happen while implementing further features) the data will stay consistent on the database.
I wonder what you and @gruvw think about this ?

The code could look like:
```suggestion
    return _firestore.runTransaction((transaction) async {
      final userDocRef = _collectionRef.doc(uid.value);

      final userDocSnap = await transaction.get(userDocRef);
      final userData = UserFirestore.fromDb(userDocSnap).data;
      final updatedData = userData.addPointstoDbData(points);
      transaction.update(userDocRef, updatedData);
    });
```

It is basically the same as yours but wrapped up in a transaction for atomicity.
Here is the documentation on firestore transaction : [https://firebase.google.com/docs/firestore/manage-data/transactions?hl=fr](url).

Note that this requires to refactor a little the `UserRepository` to keep the firestore instance in the private parameter `_firestore`. (But there is no need to change the constructor input parameters)

The constructor would simply change as follows (because `firestore` is already given as parameter to the constructor):
```dart
/* rest of constructor */, _firestore = firestore;
```

Reply:
I also agreed with this suggestion, thanks for pointing it out.